tests/test_job_tracker_v2.py

```diff
@@ -751,9 +927,65 @@ def test_parse_application_report_running(self):
             "memory": {"unit": "mb-seconds", "value": 96183879},
         }
 
-    def test_parse_application_report_empty(self):
+    def test_parse_application_response_empty(self):
         with pytest.raises(YarnAppReportParseException):
-            _ = YarnStatusGetter().parse_application_report(report="")
+            _ = YarnStatusGetter.parse_application_response(json={})
+
+    def test_get_authentication_provider_use_kerberos_on(self, monkeypatch):
+        """Simple check that the config has the desired effect: we get the expected auth object."""
+        monkeypatch.setenv("YARN_AUTH_USE_KERBEROS", "yes")
+        auth = YarnStatusGetter().get_authentication_provider()
+        assert isinstance(auth, requests_gssapi.HTTPKerberosAuth)
+
+    def test_get_authentication_provider_use_kerberos_off(self, monkeypatch):
+        """Simple check that the config has the desired effect: authentication should be None."""
+        monkeypatch.setenv("YARN_AUTH_USE_KERBEROS", "no")
+        assert YarnStatusGetter().get_authentication_provider() is None
```

Comment:
Same remark: passing `False`/`None` to the constructor makes it so that you can get rid of method `get_authentication_provider` and this test as well because it's probably already covered elsewhere.

Reply:
Done.